ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/ParticipantPruningIT.scala

```diff
@@ -694,6 +701,89 @@ class ParticipantPruningIT extends LedgerTestSuite {
     } yield ()
   })
 
+  test(
+    "PREventsByContractIdPruned",
+    "Expose both create and exercise events contract id",
+    allocate(SingleParty),
+  )(implicit ec => { case Participants(Participant(participant, party)) =>
+    def getEvents(dummyCid: Primitive.ContractId[Dummy]): Future[Int] = {
+      val request = GetEventsByContractIdRequest(dummyCid.unwrap, immutable.Seq(party.unwrap))
+      participant.getEventsByContractId(request).map(_.events.size)
+    }
+
+    for {
+      dummyCid <- participant.create(party, Dummy(party))
+      _ <- pruneToCurrentEnd(participant, party)
+      events1 <- getEvents(dummyCid)
+      exerciseCmd = participant.submitAndWaitRequest(party, dummyCid.exerciseDummyChoice1().command)
+      _ <- participant.submitAndWaitForTransactionTree(exerciseCmd)
+      events2 <- getEvents(dummyCid)
+      _ <- pruneToCurrentEnd(participant, party)
+      events3 <- getEvents(dummyCid)
+    } yield {
+      assertEquals("Expected single create event after prune", events1, 1)
+      assertEquals("Expected create and consume event before prune", events2, 2)
+      assertEquals("Expected no events following prune", events3, 0)
```

Comment:
Another interesting case to test would be

1. create contract
2. archive contract
3. prune up to and including 1
It is very similar but not quite to the scenario we have already, because here prune would come on top of existing interaction

Reply:
Test modified to include this scenario.